src/core/meshcop/commissioner.cpp

```diff
@@ -119,6 +177,9 @@ ThreadError Commissioner::AddJoiner(const Mac::ExtAddress *aExtAddress, const ch
 
         strncpy(mJoiners[i].mPsk, aPSKd, sizeof(mJoiners[i].mPsk));
         mJoiners[i].mValid = true;
+
+        SendCommissionerSet();
```

Comment:
Could we use `SuccessOrExit(error = SendCommissionerSet());`? So it could return an error if it fails to add Joiner's steering data due to some reason(for example: Invalid State).